### ISSUE TITLE
Discussion 

### ISSUE THREAD

**Issue by feibyte** [fedeoo/lighthouse-auto-fix#1]
**The purpose**

I'm building a new blog, like this one [blog](https://fedeoo.github.io/blog/).  I wanna speed up it. I could get some advice with `lighthouse`.  However, I have to optimize it by myself.  If there's a tool could do it automatically, it would be awesome.

The source code of the above blog is [here](https://github.com/fedeoo/blog/tree/gh-pages)

In order to simplify the task, We could assume our target is a static website.

**Any ideas on the collaboration, use case or solution?**

**Comment by feibyte** [comment]
@alex-e-leon  Firstly,  I think we should collect some requirements and see how many works we could do.   Then we should set up a metric and tool to collect the indicator we care.

**Comment by alex-e-leon** [comment]
Yeah - I think that's spot on - I reckon we should try to implement each of the lighthouse rules one by one - using your blog as a test ground.

How about picking one of the failing tests on your blog to start with and seeing if we can automate the fixing of it?

**Comment by feibyte** [comment]
Yeah, I am doing some research. 
Here's the audit result from [lighthouse]. [Audit result](https://github.com/fedeoo/booster/wiki/Feasible-Audit-Result)
If we're looking for a lib for manipulating DOM.  [cheerio](https://github.com/cheeriojs/cheerio) might be suitable for us.
I don't think we could see webpack support `HTML as entry` in this year.

**Comment by feibyte** [comment]
A static website has multiple pages.  But `ligthouse` just run on one page. Anyway, Let's focus on the main page first.

**Comment by alex-e-leon** [comment]
Rather than using cheerio - I usually prefer using jsdom, or even more powerful, just using headless chrome - as it gives you all the regular dom utils and a familiar API - as well as probably the best 2 html parsers + editors out there

**Comment by feibyte** [comment]
`jsdom` is more powerful. So far what we want is just manipulation of DOM, we could choose cheerio first. We could switch to `jsdom` when necessary.

**Comment by alex-e-leon** [comment]
Up to you : )
Personally though I prefer using jsdom anyway though since I can use
standard dom queries like querySelector etc, instead of learning and
getting familiar with cheerios API
